sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/DelegatingCatalogExtension.java

```diff
@@ -52,22 +53,22 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {}
 
   @Override
   public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
-    return delegate.listTables(namespace);
+    return ((TableCatalog)delegate).listTables(namespace);
```

Comment:
@cloud-fan is this downcasting in line with what you were thinking?

Reply:
this works. One suggestion to make the code work better:
```
private TableCatalog asTableCatalog() {
  return (TableCatalog)delegate;
}
```
and here can be `asTableCatalog().listTables(namespace)`. We can do the same for `asNamespaceCatalog`.